docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,319 @@
+        "org.opencontainers.image.created": "2023-11-25T10:32:54Z"
+      },
+      "artifactType": "application/vnd.oci.empty.v1+json",
+      "reference": "localhost:5000/hello@sha256:28653e2bb5b5a75393c3a8b58ed9998796299b41dc1ff1f55b9f0844ad7ba39c"
+    }
+  ]
+}
+```
+
+### oras manifest fetch
+
+Fetch a specified layer and show the formatted JSON of the fetched manifest in formatted JSON.
+
+```shell
+oras manifest fetch $REGISTRY/$REPO:$TAG --format '{{(first .Layers).Reference}}'
+```
+
+```json
+{
+    "reference": "$REGISTRY/$REPO:$manifest_digest",
+    "layers" : [
+        {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar",
+            "digest": "sha256:d2a84f4b8b650937ec8f73cd8be2c74add5a911ba64df27458ed8229da804a26",
+            "size": 12,
+            "annotations": {
+                "org.opencontainers.image.created":"2023-11-29T06:32:43Z"
+            },
+            "reference": "$REGISTRY/$REPO:$first_layer_digest",
+        },
+        {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar",
+            "digest": "sha256:d2a84f4b8b650937ec8f73cd8be2c74add5a911ba64df27458ed8229da804a26",
+            "size": 12,
+            "annotations": {
+                "org.opencontainers.image.created":"2023-11-30T06:32:43Z"
+            },
+            "reference": "$REGISTRY/$REPO:$second_layer_digest",
+        }
+    ]
+}
+```
+
+## Desired user experience for proposal 2
+
+In order to  the manifest fields in the output, format the output using the given [Go template](https://golang.org/pkg/text/template/). The keys of the returned JSON can be used as the values for the `--format` flag.
+
+### Format the output using the given Go template
+
+For example, push an artifact to a repository and filter out the value of `reference` and `artifactType` of the pushed artifact in the standard output.
+
+```shell
+oras push $REGISTRY/$REPO:$TAG --format "{{.Reference}}', '{{.ArtifactType}}"
+```
+
+```console
+"localhost:5000@sha256:85438e6598bf35057962fff34399a362d469ca30a317939427fca6b7a289e70d, "application/vnd.example+type"  
+```
+
+For example, pull a file and filter out the specified fields `mediaType`, `reference`, `size` of the pulled file in the standard output.
+
+```shell
+oras pull $REGISTRY/$REPO:$TAG --format "{{.MediaType}}, {{.Reference}}, {{.Size}}"
+```
+
+```console
+"application/vnd.oci.image.layer.v1.tar","sha256:85438e6598bf35057962fff34399a362d469ca30a317939427fca6b7a289e70d", 12
+```
+
+For example, filter out the specified annotation value of an artifact by the key name `org.opencontainers.image.created`, with the [index function](https://pkg.go.dev/text/template#pkg-functions) defined in Go template.
+
+```shell
+oras discover localhost:5000/hello:v1 --format '{{index .Manifest.Annotations "org.opencontainers.image.created"}}'
+```
+
+```console
+"2023-11-29T06:32:43Z"
+```
+
+## FAQ
+
+- Why not consider extending the existing `--output` flag to enable JSON formatted output?
```

Comment:
The ask here would be to updates the proposal for the reasoning that its because oras has commands. 
If there is enough love for --format due to similarities with other tools then that's good enough to state.  Basically when others read the proposal we don't want to set precedence that a breaking change prevents us from making the choice.

Reply:
Updated the reason in the FAQ section